runtime/sema/check_conditions.go

```diff
@@ -40,15 +43,157 @@ func (checker *Checker) visitConditions(conditions []*ast.Condition) {
 	}
 }
 
-func (checker *Checker) checkCondition(condition *ast.Condition) Type {
+func (checker *Checker) checkCondition(condition ast.Condition) Type {
```

Comment:
Not related to this pr: The return value seems to be always `nil` and never used. Maybe clean up this function to a void return, while at it.

Reply:
Removed in 43756fb470f7774748d87866ce4057470c786bae